### PR TITLE
refactor(all): align to template-formula; add clientlibs

### PR DESCRIPTION
**Pull request by noelmcloughlin**
BREAKING CHANGE: The data dictionary is simplified and expanded.
Retest your states and update pillar data accordingly.
For developer convenience, clientlibs states were introduced.
See pillar.example, defaults.yaml, and docs/README.

### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [x] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

YES

### Related issues and/or pull requests

Includes fix for
https://github.com/saltstack-formulas/prometheus-formula/issues/3
https://github.com/saltstack-formulas/prometheus-formula/issues/10
https://github.com/saltstack-formulas/prometheus-formula/issues/23
https://github.com/saltstack-formulas/prometheus-formula/issues/24

Partly addresses 
https://github.com/saltstack-formulas/prometheus-formula/issues/27

### Describe the changes you're proposing

This PR adds the following
- Better Linux support (firewall, systemd) resolving #3, #10, #23, #24
- Semantic Release
- Improved CI test coverage (Inspec)
- Improved CI test success 
- Add support for Prometheus Client Libraries
- Align to template formula

More work is needed in future PR
- bump versions
- FreeBSD support.
- MacOS support

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [x] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
Helps address https://github.com/saltstack-formulas/prometheus-formula/pull/25#issuecomment-565038325

**Comment by noelmcloughlin** [comment]
Hi @myii did you get the opportunity to run ssf against this PR?

**Comment by myii** [comment]
> Hi @myii did you get the opportunity to run ssf against this PR?

@noelmcloughlin Unfortunately, this one slipped through and wasn't even on the list in Taiga -- I'll add it now.  Please don't merge this as-is because I've noticed a few issues as I've had a quick scan.  I'll mention some of them inline shortly.

File: AUTHORS.md [hunk 1]
```diff
@@ -4,12 +4,7 @@ This list is sorted by the number of commits per contributor in _descending_ ord
```
Comment (myii) [inline AUTHORS.md:4]: This file is automatically modified so shouldn't be part of the PR.

File: CODEOWNERS [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
 
 # SECTION: Owner(s) for everything in the repo, unless a later match takes precedence
-#          **************************************************************************
-#          ***                    NO GLOBAL OWNER(S) SPECIFIED                    ***
-#          ***  Ideally this will be defined for a healthy, well-maintained repo  ***
-#          **************************************************************************
 # FILE PATTERN                              OWNER(S)
-*                                           @NONE
+*                                           @alxwr
+*                                           @noelmcloughlin
```
Comment (myii) [inline CODEOWNERS:6]: Multiple entries are handled like this:

* https://github.com/saltstack-formulas/postgres-formula/blob/1c0657b17481ad5633c69e322a48af67dc16f2f7/CODEOWNERS#L5

File: FORMULA [hunk 3]
```diff
@@ -1,9 +1,9 @@
 name: prometheus
-os: FreeBSD
-os_family: FreeBSD
-version: 3.3.0
+os: Debian, Ubuntu, Raspbian, RedHat, Fedora, CentOS, Amazon, Suse, openSUSE, Gentoo, Funtoo, Arch, Manjaro, Alpine, FreeBSD, OpenBSD, Solaris, SmartOS, Windows, MacOS
+os_family: Debian, RedHat, Suse, Gentoo, Arch, Alpine, FreeBSD, OpenBSD, Solaris, Windows, MacOS
+version: 0.2.2
```
Comment (myii) [inline FORMULA:4]: Shouldn't touch the `version` once `semantic-release` is already running for a formula.

File: docs/AUTHORS.rst [hunk 4]
```diff
@@ -15,21 +15,72 @@ This list is sorted by the number of commits per contributor in *descending* ord
      - Contributions
```
Comment (myii) [inline docs/AUTHORS.rst:15]: This file is automatically modified so shouldn't be part of the PR.

File: docs/CHANGELOG.rst [hunk 5]
```diff
@@ -1,235 +0,0 @@
-
```
Comment (myii) [inline docs/CHANGELOG.rst:1]: This file is automatically modified so shouldn't be part of the PR.

File: docs/CONTRIBUTING.rst [hunk 6]
```diff
@@ -0,0 +1,186 @@
+.. _contributing:
```
Comment (myii) [inline docs/CONTRIBUTING.rst:1]: We no longer add this to the formulas themselves, since it's available in the `.github` repo.

File: docs/CONTRIBUTING_DOCS.rst [hunk 7]
```diff
@@ -0,0 +1,96 @@
+.. _contributing_docs:
```
Comment (myii) [inline docs/CONTRIBUTING_DOCS.rst:1]: This isn't added to formula repos.

**Comment by myii** [comment]
Just a few comments above for now, I haven't been through the whole PR.

File: AUTHORS.md (hunk 1 above)
Comment (noelmcloughlin) [reply AUTHORS.md:4]: Should I remove from PR?

File: docs/CONTRIBUTING_DOCS.rst (hunk 7 above)
Comment (noelmcloughlin) [reply docs/CONTRIBUTING_DOCS.rst:1]: Should I remove the file?

File: docs/CONTRIBUTING.rst (hunk 6 above)
Comment (noelmcloughlin) [reply docs/CONTRIBUTING.rst:1]: remove?

File: docs/CHANGELOG.rst (hunk 5 above)
Comment (noelmcloughlin) [reply docs/CHANGELOG.rst:1]: remove?

File: docs/AUTHORS.rst (hunk 4 above)
Comment (noelmcloughlin) [reply docs/AUTHORS.rst:15]: remove?

File: FORMULA (hunk 3 above)
Comment (noelmcloughlin) [reply FORMULA:4]: okay, I'll fix this

File: CODEOWNERS (hunk 2 above)
Comment (noelmcloughlin) [reply CODEOWNERS:6]: okay, will fix

File: AUTHORS.md (hunk 1 above)
Comment (myii) [reply AUTHORS.md:4]: Yes, please.

File: docs/AUTHORS.rst (hunk 4 above)
Comment (myii) [reply docs/AUTHORS.rst:15]: Yes, please.

File: docs/CHANGELOG.rst (hunk 5 above)
Comment (myii) [reply docs/CHANGELOG.rst:1]: Yes, please.

File: docs/CONTRIBUTING.rst (hunk 6 above)
Comment (myii) [reply docs/CONTRIBUTING.rst:1]: Yes, please.

File: docs/CONTRIBUTING_DOCS.rst (hunk 7 above)
Comment (myii) [reply docs/CONTRIBUTING_DOCS.rst:1]: Yes, please.

**Comment by myii** [comment]
Still haven't had a chance for the proper review.  It looks like the `rubocop.yml` is an older version as well.  Perhaps you ran a diff with an older version of the `template-formula`, @noelmcloughlin?

**Comment by noelmcloughlin** [comment]
> It looks like the `bin/kitchen` is an older version

I have updated the PR for comments.

**Comment by myii** [comment]
@noelmcloughlin I'm catching up with the backlog on the `ssf-formula` at the moment.  I'll get around to this PR once I've done that.

**Comment by myii** [comment]
@noelmcloughlin Please note, there has just been an update to `prometheus/libtofs.jinja` pushed to this formula (standardised across all formulas).

**Comment by noelmcloughlin** [comment]
I fixed conflict with `prometheus/libtofs.jinja` 

selfie-merging as pr is open three months.
